datareduction/datareductionoperator.cpp

```diff
@@ -1627,6 +1626,149 @@ namespace DRO {
       return true;
    }
 
+   /*! \brief Precipitation directional differential number flux (along line)
+    * Evaluation of the precipitating differential flux (per population) targeted at low energies.
+    * In a selected number (default: 16) of logarithmically spaced energy bins, the average of
+    *      V*V/mass
+    * is calculated along (or antiparallel to) the magnetic field direction.
+    * The differential flux is converted in part. / cm^2 / s / sr / eV (unit used by observers).
+    * Parameters that can be set in cfg file under [{species}_precipitation]: nChannels, emin [eV], emax [eV]
+    * The energy channels are saved in bulk files as PrecipitationCentreEnergyLine{channel_number}.
+    */
+   VariablePrecipitationLineDiffFlux::VariablePrecipitationLineDiffFlux(cuint _popID): DataReductionOperatorHasParameters(),popID(_popID) {
+      popName = getObjectWrapper().particleSpecies[popID].name;
+      emin = getObjectWrapper().particleSpecies[popID].precipitationEmin;    // already converted to SI
+      emax = getObjectWrapper().particleSpecies[popID].precipitationEmax;    // already converted to SI
+      nChannels = getObjectWrapper().particleSpecies[popID].precipitationNChannels; // number of energy channels, logarithmically spaced between emin and emax
+      for (int i=0; i<nChannels; i++){
+         channels.push_back(emin * pow(emax/emin,(Real)i/(nChannels-1)));
```

Comment:
I know it's copied from the previous DRO but why is this with a `nChannels-1` denominator?